randovania/generator/hint_distributor.py

```diff
@@ -685,6 +692,24 @@ async def assign_precision_to_hints(
         return self.replace_hints_without_precision_with_jokes(patches)
 
 
+def _should_use_resolver_hints(config: BaseConfiguration) -> bool:
+    return config.use_resolver_hints or config.dock_rando.mode == DockRandoMode.DOCKS
+
+
+async def get_resolver_hint_state(player: int, patches: GamePatches) -> ResolverHintState | None:
+    with debug.with_level(0):
+        new_state = await resolver.resolve(patches.configuration, patches, collect_hint_data=True)
+
+    if new_state is None:
+        logger.warning(
+            f"Unable to solve game for player {player + 1} after placing doors. Hints will be based on generator state."
+        )
+        return None
+    else:
+        debug.debug_print(f">> Player {player + 1} is solve-able after door placement. Beginning hint placement.")
+        return new_state.hint_state
```

Comment:
Is it just me or is the (optional) check after generation that the seed can be solved simply duplicated work when you do this? Is there any way to skip the duplicated work?

Reply:
it is technically duplicated work, but it would be *really* ugly to try and skip it imo. frankly i'm not too concerned about adding one extra resolver run at the end of like 300 of them, lol